Makefile.example

```diff
@@ -0,0 +1,113 @@
+# Example Makefile for developer convenience
+# Based on https://gist.github.com/gambala/4874e2b41a52ac043a274a61f5d7726b
+#
+# There's nothing here you can't do the long way. This is just a collection of (hopefully intuitive) shortcuts
+# to ease common development tasks.
+#
+# To use this file:
+#
+#   $ ln -s Makefile.example Makefile
```

Comment:
Help me understand why we don't just check the full Makefile in, rather than the example?

Reply:
honestly its just the norm I'm used to following to avoid bashing folks un-checked-in personal Makefiles, and to allow people to copy & tweak vs have to use the repo make commands as is.

Happy to discard and just have the repo have a Makefile

Reply:
I think I'm pro just Doin' the Makefile - that's the pattern I've seen more than individual private makefiles, and I think we can be opinionated about this kinda thing. That said, this is your idea and I trust your judgment so go with your gut.